### ISSUE TITLE
We cannot have Kronecker product / direct product / varComb() as variance structure, right?

### ISSUE THREAD

**Issue by SchmidtPaul** [glmmTMB/glmmTMB#592]
Hi there, great work so far!
I don't think it is possible to fit variance-covariance matrices that are the Kronecker product (_a.k.a._ direct product) of mutiple terms in the model. I am only assuming this, because I could not find any hint about it in the documentation or anywhere else. 

I would be thankful, if you could confirm this and in case I am correct and maybe let me know if it is planned to include this feature.

**Here is a reproducible example:**
```{r}
dat <- agridat::mcconway.turnip %>%
  mutate(unit = 1:n() %>% as.factor,
         densf = density %>% as.factor) %>%
  mutate(date_densf = interaction(date, densf))

# nlme
mod1.nlme <- nlme::lme(
  fixed   = yield ~ gen * date * densf,  
  random  = ~ 1 | block, 
  weights = varIdent(form =  ~ 1 | date_densf),
  data    = dat)

mod2.nlme <- nlme::lme(
  fixed   = yield ~ gen * date * densf,  
  random  = ~ 1 | block, 
  weights = varComb(varIdent(form =  ~ 1 | date),
                    varIdent(form =  ~ 1 | densf)),
  data    = dat)

# glmmTMB
mod1.glmmTMB <- glmmTMB::glmmTMB(
  formula = yield ~ gen * date * densf + (1 | block) + diag(date:densf+0|unit),
  dispformula = ~ 0,   # force variance into G-side
  REML = TRUE,       
  data = dat)

# mod2.glmmTMB not possible?
```
([here are some more details](https://schmidtpaul.github.io/MMFAIR/heterogeneous_error_variance.html#5_models) on this)

glmmTMB_1.0.1 
nlme_3.1-147

Thanks in advance!

**Comment by bbolker** [comment]
You're correct that it's not available at the moment.  It shouldn't be impossible, but we'd have to think about the interface and the implementation ...

If you were thinking about digging in and trying to implement this yourself (don't know your level of interest/technical capability), the implementation of various covariance structures is in the `termwise_nll` function that starts [here](https://github.com/glmmTMB/glmmTMB/blob/master/glmmTMB/src/glmmTMB.cpp#L361) ...

**Comment by SchmidtPaul** [comment]
Thank you!
I am not sure if my technical capability is good enough, but I might give it a try in the future.

**Comment by bbolker** [comment]
I'm going to go ahead and re-open this as it's certainly a capability we'd like to add ...

**Comment by Dekermanjian** [comment]
Hello, I am wondering if there is any update on allowing Kronecker product covariance structures. This is something that would be greatly beneficial to the R programming community since currently advanced covariance structures like us@us, us@cs, and us@ar1 would need to be fit using a SAS which is not open source and not accessible to everyone.
